rdflib/namespace.py

```diff
@@ -325,12 +325,16 @@ def compute_qname(self, uri, generate=True):
         if not _is_valid_uri(uri):
             raise Exception('"%s" does not look like a valid URI, I cannot serialize this. Perhaps you wanted to urlencode it?'%uri)
 
-
         if not uri in self.__cache:
-            namespace, name = split_uri(uri)
-            namespace = URIRef(namespace)
-            prefix = self.store.prefix(namespace)
-            if prefix is None:
+            name = None
+            for lennamespace, prefix, namespace in self.ordered_prefixnames:
+                if namespace in uri:
```

Comment:
My testing showed that startswith is slower, however if there are cases where a namespace might somehow appear somewhere other than at the start of a URI then we would need to use startswith to prevent false positives. Personally I can't come up with any cases where this might happen.